articles/create-a-node-cli/index.md

```diff
@@ -0,0 +1,265 @@
+
+The command line or the terminal is a powerful tool used by a lot of developers. It allows developers to interact with the operating system using commands.These commands are passed as input strings of text.
+
+A lot of tools used by developers have a terminal CLI or command line interface. Using the command line is easy and direct hence why a lot of people prefer using it.
+
+Creating a CLI application is not a challenging task. This advantage has helped make them more popular. 
+
+Some of the most popular command line interface applications are project generators like [create-react-app](https://www.npmjs.com/package/create-react-app) for generating a React.js boilerplate application, [angular CLI](https://www.npmjs.com/package/@angular/cli) for creating an Angular.js project and [Vue CLI](https://www.npmjs.com/package/@vue/cli) for Vue.js applications.
+
+The above examples are a bit more complex in functionality and can be confusing. Yet, the basics are still the same. They all have Node.js as a common factor.
+
+Getting started with CLI applications using Node.js couldn't be easier. To tackle the basics, let's create a simple CLI application that takes two arguments `joke` or `quote` and generates a joke or a quote.
+
+## The Shebang
+
+First things first, the starting point of any CLI application point is a line called the Shebang.
+
+In Node.js the shebang looks something like this:
+
+```javascript
+#!/usr/bin/env node
+
+```
+
+The shebang tells the operating system what type of code is in the script. It also tells which interpreter it should assign to the script. If it were a python script, it would look something like this:
+
+```python
+#!/usr/bin/env py
+```
+
+The shebang, is only applicable in Unix/Linux operating systems. It should always be the first line of the entry file.
+
+### Project Setup
+
+To get started, let's setup our project and Node.js environment. In the terminal, enter:
+
+```terminal
+mkdir node-cli
+cd node-cli
+npm init -y
+```
+
+NOTE: passing the `-y` flag to npm init, will instantiate a new npm project and set everything to default and yes.
+
+This will generate our package.json file.
+
+Next up, let's install all the packages we will make use of in the project. 
+
+- [axios](https://www.npmjs.com/package/axios) for making our network requests.
+- [chalk](https://www.npmjs.com/package/chalk) for making our user interface better looking and stylish.
+- [yargs](https://www.npmjs.com/package/yargs) for all our options and argument parsing.
+
+Let's go ahead and install them:
+
+```terminal
+    npm i axios chalk yargs
+```
+
+### Getting the jokes and quotes
+
+Let's setup the jokes and quotes. We will be using a [quote-generator-api](https://quotes.rest/) and [jokes-generator-api](https://official-joke-api.appspot.com/). We will fetch the individual jokes and display them on the terminal when the user asks for them.
+
+We will create two functions that will do the fetching using axios.
+
+```javascript
```

Comment:
You should tell users they need to create a file called `index.js`.

Reply:
Can you add this, @katungi?

Reply:
Oh yeah, I hadn't pushed my changes, But I have added it.

Reply:
I have pushed the changes; I did not want to push before you explained the other comment in depth. I have pushed the changes, you can review them @louisefindlay23